### ISSUE TITLE
Slow inverse_laplace_transform

### ISSUE THREAD

**Issue by AmewsSoft** [sympy/sympy#23623]
Running an inverse Laplace Transform can take a very long time. Take the following code:

    a, t, s = sympy.symbols('a, t, s', real=True)
    
    f = -15-sympy.exp(6*t)-sympy.sin(3*t)
![image](https://user-images.githubusercontent.com/17097491/173278322-f2e6a39d-6d33-4b84-9ef3-99474a4afa93.png)

    F =  sympy.laplace_transform(f, t, s, noconds=True)
![image](https://user-images.githubusercontent.com/17097491/173278442-16815fad-6d4a-404a-be71-e94becafea39.png)

The above code completes in a few seconds but when executing the following to "undo" the previous transform:

    invF = sympy.inverse_laplace_transform(F, s, t)

The process takes many minutes (at least15 on my machine) to complete.

**Comment by oscarbenjamin** [comment]
CC @hanspi42 

I think this is because there is a rule-based implementation of the forward transform but not the inverse.

**Comment by oscarbenjamin** [comment]
It might make a difference whether you have gmpy2 installed. Running under the profiler I get:
```python
In [17]: %prun -s cumulative inverse_laplace_transform(F[0], s, t)

         8290040 function calls (7926742 primitive calls) in 4.591 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     85/1    0.006    0.000    4.592    4.592 {built-in method builtins.exec}
        1    0.000    0.000    4.592    4.592 transforms.py:2053(inverse_laplace_transform)
      4/1    0.000    0.000    4.592    4.592 transforms.py:146(doit)
      4/1    0.000    0.000    4.592    4.592 transforms.py:129(_try_directly)
        1    0.000    0.000    4.591    4.591 transforms.py:2044(_compute_transform)
      7/1    0.000    0.000    4.591    4.591 transforms.py:253(wrapper)
      4/1    0.000    0.000    4.591    4.591 transforms.py:1937(_inverse_laplace_transform)
        1    0.000    0.000    4.381    4.381 transforms.py:1963(<listcomp>)
        4    0.000    0.000    4.229    1.057 transforms.py:226(_simplify)
     29/8    0.001    0.000    4.173    0.522 simplify.py:420(simplify)
  7371/46    0.007    0.000    3.812    0.083 traversal.py:233(bottom_up)
        2    0.000    0.000    3.741    1.871 trigsimp.py:429(trigsimp)
        2    0.000    0.000    3.741    1.871 trigsimp.py:516(<lambda>)
        2    0.000    0.000    3.741    1.871 trigsimp.py:1082(futrig)
       45    0.000    0.000    3.726    0.083 trigsimp.py:1125(_futrig)
   170/17    0.000    0.000    3.722    0.219 core.py:110(minrule)
   170/17    0.000    0.000    3.711    0.218 core.py:111(<listcomp>)
       17    0.000    0.000    3.711    0.218 core.py:39(chain_rl)
      187    0.000    0.000    3.518    0.019 trigsimp.py:1198(_eapply)
       65    0.000    0.000    3.505    0.054 polytools.py:6362(factor)
       65    0.000    0.000    3.505    0.054 polytools.py:6128(_generic_factor)
       65    0.000    0.000    3.503    0.054 polytools.py:6069(_symbolic_factor)
   209/65    0.001    0.000    3.434    0.053 polytools.py:6011(_symbolic_factor_list)
   139/67    0.000    0.000    3.398    0.051 polytools.py:3335(factor_list)
   139/67    0.000    0.000    3.397    0.051 polyclasses.py:821(factor_list)
   175/67    0.001    0.000    3.396    0.051 factortools.py:1390(dmp_factor_list)
       18    0.000    0.000    3.356    0.186 factortools.py:1213(dmp_zz_i_factor)
       18    0.000    0.000    3.322    0.185 factortools.py:1202(dmp_qq_i_factor)
       18    0.000    0.000    2.982    0.166 factortools.py:1266(dmp_ext_factor)
49781/49543    0.064    0.000    2.219    0.000 decorators.py:224(_func)
  3684/88    0.003    0.000    2.134    0.024 traversal.py:241(<listcomp>)
    42375    0.134    0.000    1.724    0.000 expr.py:373(__lt__)
43262/43233    0.115    0.000    1.610    0.000 relational.py:822(__new__)
  888/252    0.001    0.000    1.597    0.006 euclidtools.py:1608(dmp_gcd)
  892/256    0.002    0.000    1.597    0.006 euclidtools.py:1560(dmp_inner_gcd)
  690/234    0.001    0.000    1.561    0.007 euclidtools.py:1524(_dmp_inner_gcd)
     5190    0.010    0.000    1.532    0.000 densearith.py:160(dmp_mul_term)
     1736    0.002    0.000    1.525    0.001 densearith.py:1608(dmp_div)
29014/6154    0.079    0.000    1.520    0.000 densearith.py:792(dmp_mul)
     5082    0.004    0.000    1.512    0.000 densearith.py:184(<listcomp>)
       54    0.001    0.000    1.459    0.027 sqfreetools.py:228(dmp_sqf_part)

```

**Comment by AmewsSoft** [comment]
Apologies, it appears this may have been an issue on my end. I did a full reinstall after reading the comments here and that seems to have resolved it. Thanks to all who replied.

**Comment by oscarbenjamin** [comment]
I'm going to reopen this because I still think it's far too slow.

**Comment by hanspi42** [comment]
I agree. I'll work on the rule-based implementation over the summer. I made several attempts already but got stuck because I already feel that the main issue will eventually be to improve the code that can do partial fraction decompositions.  So I kind of got into a "where to start" loop.  

My present plan is to start with just the rules first and then see how much work is needed for the partial fraction decomposition.

**Comment by hanspi42** [comment]
The present development state already covers the example given above ;)

**Comment by hanspi42** [comment]
> ```python
> In [17]: %prun -s cumulative inverse_laplace_transform(F[0], s, t)
> 
>          8290040 function calls (7926742 primitive calls) in 4.591 seconds
> ```

In #24579 it is 

```
         502047 function calls (458921 primitive calls) in 0.283 seconds
```

**Comment by hanspi42** [comment]
Final results as merged in PR #24579:

First run, which also needs to cache some things:
``         
696464 function calls (644634 primitive calls) in 0.825 seconds
``
Subsequent runs:
``
418197 function calls (393632 primitive calls) in 0.253 seconds
``
